crates/uv/src/commands/python/pin.rs

```diff
@@ -124,33 +130,48 @@ pub(crate) async fn pin(
         request
     };
 
-    let existing = version_file.ok().flatten();
     // TODO(zanieb): Allow updating the discovered version file with an `--update` flag.
-    let new = PythonVersionFile::new(project_dir.join(PYTHON_VERSION_FILENAME))
-        .with_versions(vec![request]);
-
-    new.write().await?;
+    if let Some(existing) = version_file {
+        if existing.contains(&request) {
+            writeln!(
+                printer.stdout(),
+                "`{}` is already pinned at `{}`",
+                existing.path().user_display().cyan(),
+                request.to_canonical_string().green(),
+            )?;
+            return Ok(ExitStatus::Success);
+        }
```

Comment:
Behavior change: `uv python pin` now cross-checks the requested version with all versions in the version file, not just the first one. It also avoids updating the version file if the required version already exists. The message updated to "`.python-version` is already pinned at `3.12`".